### ISSUE TITLE
Deploy to heroku

### ISSUE THREAD

**Issue by rfines** [KunalKapadia/express-mongoose-es6-rest-api#207]
I am trying to use this boilerplate and deploy the resulting api to Heroku but I am running into a host of problems. First among them is that no matter what I do the dist directory is not found so I am unable to run the dist/index.js file even after removing the dist/ directory from the .gitignore.

I am currently seeing this in the logs:

    Starting process with command `node dist/default` 
  module.js:471 
  Process exited with status 1 
      throw err; 
      ^ 
     Error: Cannot find module '/app/dist/default' 
      at Function.Module._resolveFilename (module.js:469:15) 
      at Function.Module._load (module.js:417:25) 
      at Module.runMain (module.js:604:10) 
      at run (bootstrap_node.js:394:7) 
      at startup (bootstrap_node.js:149:9) 
      at bootstrap_node.js:509:3 

Can anybody help me?

**Comment by simonschllng** [comment]
Could you please share your solution if this is really solved?

**Comment by rfines** [comment]
Well the biggest problem was my build process for heroku was wiping out the dist/ directory but I wasn't running the gulp build task in my heroku:production task so the project was never getting recompiled. 
In addition I had to copy all of the devDependencies to the dependencies in order for my gulp tasks to run without blowing up. 
I also had to add a simple Procfile to the root of the project with the following line in it:
      
    web: node dist/index.js

to serve the site through express.
Finally the line in index.js where the app listens on a port using config.port need to be changed to listen to process.env.port for anything other than development configs.

It is also worth noting that, for Heroku at least, you cannot follow the deployment steps that are listed in the readme because you cannot scp files to a Heroku server. I currently have the entire app pushed to the heroku repo but I plan on changing it so that the dist directory is its own repo that is managed separately.
